R/create_table_patient_data_changes_only.R

```diff
@@ -0,0 +1,97 @@
+#' Group this data by id and take only the months when there is a change in the medical data.
+#'
+#'
+#' @param patient_data_files list of CSV files with cleaned patient data from step 2.
+#' @param input_root root directory of the input CSV files.
+#' @param output_root root directory of the output folder.
+#' @param variable name of the column that should be exported.
+#' @param name name used to create the export file name.
+create_table_longitudinal_data <-
+    function(patient_data_files,
+             input_root,
+             output_root,
+             variable,
+             name) {
+        logInfo("Start creating single csv for table create_table_longitudinal_data and variable ", variable)
+
+        dynamic_patient_columns <-
+            c(
+                "blood_pressure_dias_mmhg",
+                "blood_pressure_sys_mmhg",
+                "bmi",
+                "bmi_date",
+                "clinic_code",
+                "country_code",
+                "fbg_updated_date",
+                "fbg_updated_mg",
+                "fbg_updated_mmol",
+                "file_name",
+                "hba1c_updated",
+                "hba1c_updated_exceeds",
+                "hba1c_updated_date",
+                "height",
+                "hospitalisation_cause",
+                "hospitalisation_date",
+                "id",
+                "insulin_regimen",
+                "last_clinic_visit_date",
+                "last_remote_followup_date",
+                "observations",
+                "observations_category",
+                "sheet_name",
+                "status",
+                "support_from_a4d",
+                "testing_frequency",
+                "tracker_month",
+                "tracker_year",
+                "updated_2022_date",
+                "weight"
+            )
+
+        patient_data_list <- list()
+
+        # get the latest static patient data for each tracker file
+        for (patient_file in patient_data_files) {
+            patient_data <- read_csv(
+                file.path(input_root, patient_file),
+                locale = readr::locale(encoding = "UTF-16LE"),
+                show_col_types = F,
+                col_types = "iiinDccDcnnDnncnlnlDncDccDDDccccDccccciDciiiDn",
+                col_select = all_of(dynamic_patient_columns)
+            )
+
+            patient_data_list[[patient_file]] <- patient_data
+        }
+
+        # Complete dataframe
+        patient_data_df <- patient_data_list %>%
+            bind_rows()
+
+        # get latest static patient data overall
+        variable_lag <- paste0(variable, "_lag")
+        longitudinal_data <- patient_data_df %>%
+            drop_na(!!variable) %>%
+            dplyr::filter(!!variable != ERROR_VAL_NUMERIC) %>%
+            group_by(id) %>%
+            arrange(tracker_year, tracker_month) %>%
+            mutate(!!variable_lag := dplyr::lag(get(variable), default = NULL)) %>%
```

Comment:
Why use `:=`? Doesn't seem to be a standard use case. Why not just use `=`instead?

Reply:
As I said I just googled how to use dynamic variables and !! Is always mentioned with := to resolve that variable correctly so if you know how it works correctly or how to better achieve this dynamic behavior I am very open

Reply:
Ok its all fine. I looked a little closer at the rlang documentation and in the context of the newer glue notation (stuff like `mutate("{variable}_lag" := ...)` see [here](https://rlang.r-lib.org/reference/args_data_masking.html)) they linked an issue that wanted to get rid of `:=` but the issue was actually closed with the comment that this is fundamentally not possible. So we won't get rid of `:=` in mutate.

Reply:
`     df %>% mutate(dplyr::across(.cols = {{var}}, .fns=function(x) {dplyr::lag(x, n = 1) }, .names =  "{.col}_lag"
`

I came across across, recently  ;).  Its pretty powerful.